### PR TITLE
MICROS-6921 Fix pod disruption budget for low minReplicas

### PR DESCRIPTION
**Pull request by halcyonCorsair**
If there are < 3 replicas, the kubecompute pod disruption budget must be raised accordingly to prevent node drain issues.

### CODE REVIEW

File: pkg/orchestration/wiring/k8scompute/plugin.go [hunk 1]
```diff
@@ -539,11 +537,22 @@ func buildAntiAffinity(labelMap map[string]string) *core_v1.PodAntiAffinity {
 	}
 }
 
-func buildPodDisruptionBudgetSpec(labelMap map[string]string) policy_v1.PodDisruptionBudgetSpec {
+func buildPodDisruptionBudgetSpec(labelMap map[string]string, minReplicas int32) policy_v1.PodDisruptionBudgetSpec {
+	// Make sure we can drain nodes if the minReplicas is < 3
+	var pdbPercentage string
+	switch minReplicas {
+	case 0, 1:
+		pdbPercentage = "100%"
```
Comment (patrickshan) [inline pkg/orchestration/wiring/k8scompute/plugin.go]: should be 0 here since it's "MinAvailable" ?

Comment (halcyonCorsair) [reply pkg/orchestration/wiring/k8scompute/plugin.go]: Whoops, yes percentages were inverted.

Comment (halcyonCorsair) [reply pkg/orchestration/wiring/k8scompute/plugin.go]: Fixed now.

**Comment by amckague** [comment]
I've removed red members from this PR. This sort of change (twiddling the output number) should be ok to go through without our team directly shepherding it.

File: pkg/orchestration/wiring/k8scompute/plugin.go [hunk 2]
```diff
@@ -539,11 +537,22 @@ func buildAntiAffinity(labelMap map[string]string) *core_v1.PodAntiAffinity {
...
+func buildPodDisruptionBudgetSpec(labelMap map[string]string, minReplicas int32) policy_v1.PodDisruptionBudgetSpec {
+	// Make sure we can drain nodes if the minReplicas is < 3
+	var pdbPercentage string
+	switch minReplicas {
+	case 0, 1:
+		pdbPercentage = "0%"
+	case 2:
+		pdbPercentage = "50%"
+	default:
+		pdbPercentage = "66%"
```
Comment (jokeyrhyme) [inline pkg/orchestration/wiring/k8scompute/plugin.go]: Is it okay for more than half the Pods to be disrupted? Should we maybe have 33% for replicas >= 3 ? Or maybe even make the `case 2` the default, for 50% with replicas >= 2 ?
If these were EC2s, we'd be this conservative, but I don't know if we can be more disruptive with Pods, maybe?

Comment (halcyonCorsair) [reply pkg/orchestration/wiring/k8scompute/plugin.go]: This is a poorly named variable.  I made this mistake myself.  We are configuring "minimum available" here, eg. we want at least 2/3rds of the nodes to be available when > 3 nodes.

I'll rename the variable to make this more clear.

**Pull request merged**